x-pack/plugin/profiling/src/test/java/org/elasticsearch/xpack/profiling/HostMetadataTests.java

```diff
@@ -38,4 +37,109 @@ public void testCreateFromRegularSource() {
         assertEquals(region, host.instanceType.region);
         assertEquals(instanceType, host.instanceType.name);
     }
+
+    public void testCreateFromSourceGCP() {
+        final String hostID = "1440256254710195396";
+        final String machine = "x86_64";
+        final String provider = "gcp";
+        final String[] regions = { "", "", "europe-west1", "europewest", "europe-west1" };
+        final String[] zones = {
+            "",
+            "/",
+            "projects/123456789/zones/" + regions[2] + "-b",
+            "projects/123456789/zones/" + regions[3],
+            "projects/123456789/zones/" + regions[4] + "-b-c",
+            randomAlphaOfLength(10) // keep the random value as last array entry
+        };
+
+        for (int i = 0; i < regions.length; i++) {
+            String region = regions[i];
+            String zone = zones[i];
+
+            // tag::noformat
+            HostMetadata host = HostMetadata.fromSource(
+                Map.of(
+                    "host.id", hostID,
+                    "profiling.host.machine", machine,
+                    "gce.instance.zone", zone
+                )
+            );
+            // end::noformat
+
+            assertEquals(hostID, host.hostID);
+            assertEquals(machine, host.profilingHostMachine);
+            assertEquals(provider, host.instanceType.provider);
+            if (i < zones.length - 1) {
```

Comment:
I suppose that is meant to exclude the assert when we have random garbage in `zone`? Shouldn't `zone` == `region` in this case? Can we expand the assert accordingly and add a comment? If you prefer, you could also extract this into its own test case.

Reply:
> If you prefer, you could also extract this into its own test case.

I didn't do this on purpose to avoid code duplication. But my mind just changed when rethinking this, see below.

> Shouldn't zone.equals(region) evaluate to true in this case?

Yes with only alphanumeric random strings.

IMO, we should not use a random test, but rather "fuzz" all the combinations of relevant characters. That is `/`, `-` and <any other character>. Testing all combinations of a 0-5 characters long string completely covers the parsing code, where each character can be any of (`/`. `-`, `a`). This avoids random test failures (which can be tedious for all other developers) and should be fast enough to not add relevant latency to the test suite.